src/ddspdrum/module.py

```diff
@@ -58,6 +62,94 @@ def control_to_sample_rate(self, control: np.array) -> np.array:
+        scale (float)       :   scaling when converting between [0,1] range
+        """
+        if parameter_id in self.parameters:
+            raise ValueError("parameter_id: {} already used".format(parameter_id))
+
+        self.parameters[parameter_id] = Parameter(value, minimum, maximum,
+                                                  scale, parameter_id)
+
+    def connect_parameter(self, parameter_id: str, module: SynthModule,
+                          module_parameter_id: str):
+        """
+        Create a named parameter for this synthesizer that is connected to a parameter
+        in an underlying synth module
+
+        Parameters
+        ----------
+        parameter_id (str)          : name of the new parameter
+        module (SynthModule)        : the SynthModule to connect to this parameter
+        module_parameter_id (str)   : parameter_id in SynthModule to target
+        """
+        if parameter_id in self.parameters:
+            raise ValueError("parameter_id: {} already used".format(parameter_id))
+
+        if module_parameter_id not in module.parameters:
+            raise KeyError("parameter_id: {} not a parameter in {}".format(
+                module_parameter_id, module))
+
+        self.parameters[parameter_id] = module.get_parameter(module_parameter_id)
+
+    def get_parameter(self, parameter_id: str) -> Parameter:
+        """
+        Get a single parameter for this module
+
+        Parameters
+        ----------
+        parameter_id (str)  :   Id of the parameter to return
+        """
+        return self.parameters[parameter_id]
+
+    def get_parameter_0to1(self, parameter_id: str) -> float:
+        """
+        Get the value of a single parameter in the range of [0,1]
+
+        Parameters
+        ----------
+        parameter_id (str)  :   Id of the parameter to return the value for
+        """
+        return self.parameters[parameter_id].get_value_0to1()
+
+    def set_parameter(self, parameter_id: str, value: float):
+        """
+        Update a specific parameter value, ensuring that it is within a specified range
+
+        Parameters
+        ----------
+        parameter_id (str)  : Id of the parameter to update
+        value (float)       : Value to update parameter with
+        """
+        self.parameters[parameter_id].set_value(value)
+
+    def set_parameter_0to1(self, parameter_id: str, value: float):
+        """
+        Update a specific parameter with a value in the range [0,1]
+
+        Parameters
+        ----------
+        parameter_id (str)  : Id of the parameter to update
+        value (float)       : Value to update parameter with
+        """
+        self.parameters[parameter_id].set_value_0to1(value)
+
+    def list_parameters(self):
+        """
+        Return a dictionary of the parameters as strings
+        """
+        return {key: str(self.parameters[key]) for key in self.parameters}
```

Comment:
maybe return it as a dict so it can be manipulated by API?

Reply:
Idea here is to be able to get a quick look at what the parameters are, if you check out the example you get a nice print out of the parameter values / min / max in Jupyter -- maybe this would make more sense to be renamed print_parameter or something like that? Parameters can be accessed as a dict directly through the parameters member.

Reply:
Sure, maybe this should just be the `__repr__` for the class instance? The long complete string.

https://stackoverflow.com/questions/1436703/difference-between-str-and-repr

* __repr__ goal is to be unambiguous
* __str__ goal is to be readable

Reply:
Looks like `__repr__` is intended for debugging / logging and is supposed to show everything in class? Intention of this function is to get a quick snapshot into the parameters and their names for end-users. If I'm working with a SynthModule in a Jupyter notebook and I want to see what parameters are available and their range then I could call this function. Do you think `__repr__` is okay for that?

Reply:
For me __repr__ seems like the natural choice, but if you are uneasy about it or think it will involve a lot more chatty text, then we can skip it.